src/performance/metrics/StatisticCollector.cpp

```diff
@@ -46,64 +48,172 @@ int StatisticCollector::getMemoryUsageByProcess() {
+}
 
-        std::stringstream stream;
-        stream << std::fixed << std::setprecision(2) << totalCPUUsage;
-        std::string cpuUsageString = stream.str();
+long StatisticCollector::getRXBytes() {
+    FILE *file = fopen("/sys/class/net/eth0/statistics/rx_bytes", "r");
+    long result = -1;
+    fscanf(file, "%li", &result);
+    fclose(file);
+    // std::cout << "Total read bytes: " + std::to_string(result) << std::endl;
+    return result;
+}
+
+long StatisticCollector::getTXBytes() {
+    FILE *file = fopen("/sys/class/net/eth0/statistics/tx_bytes", "r");
+    long result = -1;
+    fscanf(file, "%li", &result);
+    fclose(file);
+    // std::cout << "Total sent bytes: " + std::to_string(result) << std::endl;
+    return result;
+}
 
-        vmLevelStatistics = std::to_string(totalMemoryUsed) + "," + cpuUsageString + ",";
+int StatisticCollector::getSocketCount() {
+    DIR *d = opendir("/proc/self/fd");
+    if (!d) {
+        puts("Error opening directory /proc/self/fd");
+        return -1;
+    }
+    const struct dirent *dir;
+    char path[64];
+    char link_buf[1024];
+    int count = 0;
+    while ((dir = readdir(d)) != NULL) {
+        const char *filename = dir->d_name;
+        if (filename[0] < '0' || '9' < filename[0]) continue;
+        sprintf(path, "/proc/self/fd/%s", filename);
+        size_t len = readlink(path, link_buf, sizeof(link_buf) - 1);
+        link_buf[len] = 0;
+        if (len > 0 && strncmp("socket:", link_buf, 7) == 0) {
+            count++;
+        }
     }
+    (void)closedir(d);
 
-    if (isTotalAllocationRequired == "true") {
-        long totalMemory = getTotalMemoryAllocated();
-        int totalCoresAvailable = getTotalNumberofCores();
+    // std::cout << "Total sockets: " + std::to_string(count) << std::endl;
+    return count;
+}
+
+static long parseLine(char *line) {
+    int i = strlen(line);
+    const char *p = line;
+    while (*p < '0' || *p > '9') p++;
+    line[i - 3] = '\0';
+    long val = strtol(p, NULL, 10);
+    if (val < 0 || val > 0xfffffffffffffffL) return -1;
+    return val;
+}
 
-        vmLevelStatistics = vmLevelStatistics + std::to_string(totalMemory) + "," + std::to_string(totalCoresAvailable);
+static void getCpuCycles(long long *totalp, long long *idlep) {
+    *totalp = 0;
+    *idlep = 0;
+    FILE *fp = fopen("/proc/stat", "r");
+    if (!fp) return;
+    char line[1024];
+    fscanf(fp, "%[^\r\n]%*c", line);
+    fclose(fp);
+    std::cout << line << std::endl;
+    char *p = line;
+    while (*p < '0' || *p > '9') p++;
+    long long total = 0;
+    long long idle = 0;
+    char *end_ptr = p;
+    for (int field = 1; field <= 10; field++) {
+        while (*p < '0' || *p > '9') {
+            if (!(*p)) break;
+            p++;
+        }
+        if (!(*p)) break;
+        long long value = strtoll(p, &end_ptr, 10);
+        p = end_ptr;
+        if (value < 0) {
+            std::cerr << "Value is " << value << " for line " << line << std::endl;
```

Comment:
If this is an error lets use standard logger with error()

Reply:
Done